lib/common/query-params.js

```diff
@@ -0,0 +1,248 @@
+import { getCurrentDate, getCurrentTime } from '../util/time'
+import { isTransit, isAccessMode, isCar } from '../util/itinerary'
+
+/**
+ * name: the default name of the parameter used for internal reference and API calls
+ * planTypes: array of plan type(s) (ITINERARY, PROFILE, or both) this param applies to
+ * default: the default value for this param
+ * itineraryRewrite: an optional function for translating the value for the plan API only
+ * profileRewrite: an optional function for translating the value for the profile API only
+ * label: a text label for for onscreen display
+ * selector: the default type of UI selector to use in the form. Can be one of:
+ *   - DROPDOWN: a standard drop-down menu selector
+ * options: an array of text/value pairs used with a dropdown selector
+ */
+
+const queryParams = [
+  { /* from - the trip origin. stored internally as a location (lat/lon/name) object  */
+    name: 'from',
+    planTypes: [ 'ITINERARY', 'PROFILE' ],
+    default: null,
+    itineraryRewrite: value => ({ fromPlace: `${value.lat},${value.lon}` }),
+    profileRewrite: value => ({ from: { lat: value.lat, lon: value.lon } })
```

Comment:
Both of these will throw an error if a user sends `null`, `undefined` or a number as the value.  It would be good to have some kind of validation for what shape the expected value should be.